lib/smart_proxy_main.rb

```diff
@@ -43,6 +43,7 @@
 
 module Proxy
   SETTINGS = Settings.load_global_settings
+  SETTINGS.apply_argv(ARGV)
```

Comment:
More of a nitpick: I think Settings::Global#apply_argv call could be made from Settings.load_global_settings, mostly to keep settings initialization logic in one spot. If so, Settings.load_global_settings should probably be renamed to something like Settings.initialize_global_settings...